pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java

```diff
@@ -430,22 +430,7 @@ protected CompletableFuture<Messages<T>> internalBatchReceiveAsync() {
         CompletableFuture<Messages<T>> result = cancellationHandler.createFuture();
         internalPinnedExecutor.execute(() -> {
             if (hasEnoughMessagesForBatchReceive()) {
-                MessagesImpl<T> messages = getNewMessagesImpl();
-                Message<T> msgPeeked = incomingMessages.peek();
-                while (msgPeeked != null && messages.canAdd(msgPeeked)) {
-                    Message<T> msg = incomingMessages.poll();
-                    if (msg != null) {
-                        decreaseIncomingMessageSize(msg);
```

Comment:
Looks like we will invoke `messageProcessed` after this change, and we will add this message to 
`unAckedMessageTracker`. is it what we expected?

Reply:
Yes, before logic maybe has also has bugs. I will add a new unit test cover it.

Reply:
@mattisonchao Add new unit test `testBatchReceiveAckTimeout` to cover this case. PTAL.